plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -104,19 +105,39 @@
      */
     private final String[] constructorParamNames;
 
-    /** binds type parameter */
-    static <T> DescribableModel<T> of(Class<T> clazz) {
-        return new DescribableModel<T>(clazz);
+    /** Binds type parameter, preferred means of obtaining a DescribableModel. */
+    public static <T> DescribableModel<T> of(Class<T> clazz) {
+        DescribableModel mod = modelCache.get(clazz.getName());
+        if (mod != null) {
+            return mod;
+        }
+        mod = new DescribableModel<T>(clazz);
+        modelCache.put(clazz.getName(), mod);
+        return mod;
     }
 
+    /** Map class name to cached model. */
+    static ConcurrentHashMap<String, DescribableModel> modelCache = new ConcurrentHashMap<String, DescribableModel>();
```

Comment:
Have you measured how much memory this will eat on a typical system? Perhaps using a soft reference could ease some memory?

Reply:
Good point, I'd considered it briefly, but let's spin that out a bit more.  We get one per `Class` we created a `DescribableModel` for, so a couple thousand of them at most.   The `String` keys are class names, which are cached (so it's only a pointer to a `String` we'd already have in-memory).  

With G1GC string deduplication on, any duplicated strings will have their backing `char[]` deduplicated.  This should help with Parameter names as well.  

Should be a few MB at most, I think -- and it tremendously reduces resource use by pipelines, improving runtime and garbage objects generated.